x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTimeParseProcessor.java

```diff
@@ -6,86 +6,110 @@
 package org.elasticsearch.xpack.sql.expression.function.scalar.datetime;
 
 import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.xpack.ql.expression.gen.processor.Processor;
 import org.elasticsearch.xpack.sql.SqlIllegalArgumentException;
 
+
 import java.io.IOException;
 import java.time.DateTimeException;
 import java.time.LocalDateTime;
-import java.time.ZoneOffset;
+import java.time.LocalTime;
+import java.time.OffsetTime;
 import java.time.ZonedDateTime;
+import java.time.ZoneOffset;
 import java.time.format.DateTimeFormatter;
 import java.time.temporal.TemporalAccessor;
 import java.util.Locale;
 import java.util.Objects;
+import java.util.function.BiFunction;
 
 import static org.elasticsearch.xpack.ql.util.DateUtils.UTC;
 
 public class DateTimeParseProcessor extends BinaryDateTimeProcessor {
 
-    public static final String NAME = "dtparse";
+    public enum DateTimeParseExtractor {
```

Comment:
I would rename it to `DateTimeParser` or plain `Parser` since it's already under a class so it would be: `DateTimeParseProcessor.Parser`.